Vague|Nonexistent Build Documentation

As a fresh new user, I tried to get and build this project. 

My first issue is that documentation on where/how one is supposed to download this projects is nonexistent. 

These DON'T work
```
go get github.com/kubernetes/node-problem-detector
go get k8s.io/node-problem-detector
```

Also git cloning this project anywhere except $GOPATH/src/k8sio/|$GOROOT/src/k8sio/ fails to build 
```
[user@box node-problem-detector]$ make
CGO_ENABLED=1 GOOS=linux go build -o bin/node-problem-detector \
     -ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.4.0-22-gffc7909' \
     -tags journald cmd/node_problem_detector.go
cmd/node_problem_detector.go:29:2: cannot find package "k8s.io/node-problem-detector/cmd/options" in any of:
        /home/user/go/src/node-problem-detector/vendor/k8s.io/node-problem-detector/cmd/options (vendor tree)
        /usr/lib/golang/src/k8s.io/node-problem-detector/cmd/options (from $GOROOT)
        /home/user/go/src/k8s.io/node-problem-detector/cmd/options (from $GOPATH)
cmd/node_problem_detector.go:30:2: cannot find package "k8s.io/node-problem-detector/pkg/problemclient" in any of:
        /home/user/go/src/node-problem-detector/vendor/k8s.io/node-problem-detector/pkg/problemclient (vendor tree)
        /usr/lib/golang/src/k8s.io/node-problem-detector/pkg/problemclient (from $GOROOT)
        /home/user/go/src/k8s.io/node-problem-detector/pkg/problemclient (from $GOPATH)
cmd/node_problem_detector.go:31:2: cannot find package "k8s.io/node-problem-detector/pkg/problemdetector" in any of:
        /home/user/go/src/node-problem-detector/vendor/k8s.io/node-problem-detector/pkg/problemdetector (vendor tree)
        /usr/lib/golang/src/k8s.io/node-problem-detector/pkg/problemdetector (from $GOROOT)
        /home/user/go/src/k8s.io/node-problem-detector/pkg/problemdetector (from $GOPATH)
cmd/node_problem_detector.go:32:2: cannot find package "k8s.io/node-problem-detector/pkg/systemlogmonitor" in any of:
        /home/user/go/src/node-problem-detector/vendor/k8s.io/node-problem-detector/pkg/systemlogmonitor (vendor tree)
        /usr/lib/golang/src/k8s.io/node-problem-detector/pkg/systemlogmonitor (from $GOROOT)
        /home/user/go/src/k8s.io/node-problem-detector/pkg/systemlogmonitor (from $GOPATH)
cmd/node_problem_detector.go:33:2: cannot find package "k8s.io/node-problem-detector/pkg/version" in any of:
        /home/user/go/src/node-problem-detector/vendor/k8s.io/node-problem-detector/pkg/version (vendor tree)
        /usr/lib/golang/src/k8s.io/node-problem-detector/pkg/version (from $GOROOT)
        /home/user/go/src/k8s.io/node-problem-detector/pkg/version (from $GOPATH)
make: *** [bin/node-problem-detector] Error 1

```

After locating where it expects to be cloned, making fails anyway. I guess this expects ubuntu, but this also ins't documented anywhere.
```
[user@box node-problem-detector]$ make
CGO_ENABLED=1 GOOS=linux go build -o bin/node-problem-detector \
     -ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.4.0-22-gffc7909' \
     -tags journald cmd/node_problem_detector.go
# k8s.io/node-problem-detector/vendor/github.com/coreos/go-systemd/sdjournal
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:27:33: fatal error: systemd/sd-journal.h: No such file or directory
 // #include <systemd/sd-journal.h>
                                 ^
compilation terminated.
make: *** [bin/node-problem-detector] Error 2
```

--- a/README.md
+++ b/README.md
@@ -73,9 +73,20 @@ Refer [heapster docs](https://github.com/kubernetes/heapster/blob/master/docs/so
 * `--hostname-override`: A customized node name used for node-problem-detector to update conditions and emit events. node-problem-detector gets node name first from `hostname-override`, then `NODE_NAME` environment variable and finally fall back to `os.Hostname`.
 
 ## Build Image
-Run `make` in the top directory. It will:
-* Build the binary.
-* Build the docker image. The binary and `config/` are copied into the docker image.
+* `go get` or `git clone` node-problem-detector repo into `$GOPATH/src/k8s.io` or `$GOROOT/src/k8s.io` 
+with one of the below directions:
+  * `cd $GOPATH/src/k8s.io && git clone git@github.com:kubernetes/node-problem-detector.git`
+  * `cd $GOROOT/src/k8s.io && git clone git@github.com:kubernetes/node-problem-detector.git`
+  * `cd $GOPATH/src/k8s.io && go get k8s.io/node-problem-detector`
+
+* run `make` in the top directory. It will:
+  * Build the binary.
+  * Build the docker image. The binary and `config/` are copied into the docker image.
+
+**Note**:
+By default node-problem-detector will be built with systemd support with `make` command. This requires systemd develop files. 
+You should download the systemd develop files first. For Ubuntu, `libsystemd-journal-dev` package should 
+be installed.
 
 ## Push Image
 `make push` uploads the docker image to registry. By default, the image will be uploaded to

